ionc/ion_writer_text.c

```diff
@@ -1181,7 +1259,10 @@ iERR _ion_writer_text_append_escape_sequence_string(ION_STREAM *poutput, BYTE *c
 
     c = *cp;
     if (c < 32 || c == '\\' || c == '"' || c == '\'') {
-        image = _ion_writer_get_control_escape_string(c);
+        if (!down_convert)
+            image = _ion_writer_get_control_escape_string(c);
+        else
+            image = _ion_writer_get_control_escape_string_json(c);
```

Comment:
I have a minor preference to swap the bodies and lose the negation. Disregard if this style was chosen deliberately.

Reply:
Yea, that wasn't done for any particular reason, definitely reads better without the negation.